### PR TITLE
Fix GitHub Actions YAML syntax

### LINKED ISSUE #23784: Failed docs preview actions
The actions queue is full of failed docs preview runs like this:
https://github.com/sympy/sympy/actions/runs/2682409834

I think that the trigger for the docs preview is too sensitive:
https://github.com/sympy/sympy/blob/64405a4fcb9010c95617c5e8d97aa0c142e895b9/.github/workflows/docs-preview.yml#L2

### PR DESCRIPTION
**Pull request by asmeurer**
Followup of https://github.com/sympy/sympy/pull/23788
Fixes https://github.com/sympy/sympy/issues/23784

#### Release Notes

NO ENTRY

### CODE REVIEW

File: .github/workflows/docs-preview.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Docs Preview
 on: [status]
 jobs:
   circleci_artifacts_redirector_job:
-    if: github.event.context == 'ci/circleci: Build Docs Preview'
+    if: "github.event.context == 'ci/circleci: Build Docs Preview'"
```
Comment (oscarbenjamin) [inline .github/workflows/docs-preview.yml:5]: I haven't seen this syntax before. Usually the syntax is
```
if: ${{ github.event.context == ... }}
```

Comment (asmeurer) [reply .github/workflows/docs-preview.yml:5]: I think maybe the `if` key evaluates itself as an expression automatically. The braces aren't there in the examples https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#status

Comment (oscarbenjamin) [reply .github/workflows/docs-preview.yml:5]: So what does wrapping the expression in quotes do?

Comment (asmeurer) [reply .github/workflows/docs-preview.yml:5]: Makes it valid YAML syntax. YAML usually parses things as a string unless it looks like something else. In this case, I think the `:` confuses it.

Comment (oscarbenjamin) [reply .github/workflows/docs-preview.yml:5]: Okay, that makes sense. It's supposed to a YAML string but then GA will evaluate that string as an expression.

**Pull request merged**